rye/src/cli/publish.rs

```diff
@@ -77,17 +77,16 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
     // c. Otherwise prompt for token and provide encryption option, storing the result in credentials.
     let repository = &cmd.repository;
     let mut credentials = get_credentials()?;
-    credentials
+    let repo_credentials = credentials
```

Comment:
This is a bit tangential, but this lets us pass less info to `secret_from_token`, just the repo credentials rather than the whole thing.

Happy to revert if unwanted, but it does remove a bunch of extra `gets` to this value, and gives us a proof that the key exists, instead of having to handle the `Option` returned by `.get(repository)`